File: Remove_Duplicate_Letters.c
char* removeDuplicateLetters(char* s) {
    int n=strlen(s);
    // for(int i=0;i<n-1;i++)
    // {
    //     for(int j=1+i;j<n;j++)
    //     {
    //         if(s[i]==s[j])
    //         {
    //             for(int k=j;k<n;k++)
    //             {
    //                 s[k]=s[k+1];
    //                 n--;
    //                 j--;
    //             }
    //         }
    //     }
    // }
    for(int i=n-1;i>0;i--)
    {
        for(int j=n;j>i+1;j--)
        {
            if(s[i]==s[j])
            {
                for(int k=n;k>j;k--)
                {
                    s[k+1]=s[k];
                    n--;
                    j--;
                }
            }
        }
    }
    // for(int i=0;i<n-1;i++)
    // {
    //     for (int j=i+1;j<n;j++)
    //     {
    //         if(s[i]>s[j])
    //         {
    //             char temp=s[i];
    //             s[i]=s[j];
    //             s[j]=temp;
    //         }
    //     }
    // }
    // for(int i = 0; i < n; i++) {
    //     s[i] = s[n - i - 1];
    // }
    // s[n] = '\0';
    s[n]='\0';
    return s;
}

File: Remove_Duplicate_Letters.py
class Solution:
    def removeDuplicateLetters(self, s: str) -> str:
        # x=len(s)
        # for i in range (x):
        #     for j in range (x-1):
        #         if s[i]==s[j]:
        #             for k in range (x-1-1):
        #                 s[k]=s[k+1]
        # s=sort(s)
        # return s
       
        if not s:
            return ""

        pos = 0

        for i in range(len(s)):
            if s[i] < s[pos]:
                pos = i
            if s[i] not in s[i+1:]:
                break

        first = s[pos]
        remaining = s[pos+1:].replace(first, "")

        return first + self.removeDuplicateLetters(remaining)
